modules/eventlog/src/test/java/org/apache/ignite/internal/eventlog/impl/WebhookSinkTest.java

```diff
@@ -0,0 +1,223 @@
+                                .withPath("/api/v1/events")
+                                .withContentType(APPLICATION_JSON)
+                                .withHeader("X-SINK-CLUSTER-ID", CLUSTER_ID.toString())
+                                .withHeader("X-SINK-NODE-NAME", "default")
+                                .withBody(json(expectedSentContent, ONLY_MATCHING_FIELDS)),
+                        VerificationTimes.exactly(1));
+    }
+
+    @Test
+    void shouldSendEventsByTimeout() {
+        clientAndServer
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withPath("/api/v1/events")
+                )
+                .respond(response(null));
+
+        WebhookSink sink = createSink(c -> c.changeBatchSendFrequency(200L));
+
+        sink.write(IgniteEvents.USER_AUTHENTICATION_SUCCESS.create(EventUser.of("user1", "basicProvider")));
+
+        Awaitility.await()
+                .atMost(Duration.ofMillis(500L))
+                .until(() -> sink.getLastSendMillis() > 0L);
+
+        var expectedSentContent = "[{\"type\" : \"USER_AUTHENTICATION_SUCCESS\"}]";
+
+        clientAndServer
+                .verify(
+                        request()
+                                .withMethod("POST")
+                                .withPath("/api/v1/events")
+                                .withContentType(APPLICATION_JSON)
+                                .withHeader("X-SINK-CLUSTER-ID", CLUSTER_ID.toString())
+                                .withHeader("X-SINK-NODE-NAME", "default")
+                                .withBody(json(expectedSentContent, ONLY_MATCHING_FIELDS)),
+                        VerificationTimes.exactly(1));
+    }
+
+    @Test
+    void shouldSkipSendingEvents() throws InterruptedException {
+        clientAndServer
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withPath("/api/v1/events")
+                )
+                .respond(response(null));
+
+        WebhookSink sink = createSink(c -> c.changeBatchSize(2));
+
+        sink.write(IgniteEvents.USER_AUTHENTICATION_SUCCESS.create(EventUser.of("user1", "basicProvider")));
+
+        Thread.sleep(500L);
+
+        assertThat(sink.getEvents(), hasSize(1));
+
+        clientAndServer
+                .verify(
+                        request()
+                                .withMethod("POST")
+                                .withPath("/api/v1/events")
+                                .withContentType(APPLICATION_JSON)
+                                .withHeader("X-SINK-CLUSTER-ID", CLUSTER_ID.toString())
+                                .withHeader("X-SINK-NODE-NAME", "default"),
+                        VerificationTimes.never());
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {429, 502, 503, 504})
+    void shouldTryToResendEvents(int statusCode) {
```

Comment:
Can you add one more test that shows how do we handle other error codes like 500, 400, 404, 301, etc.

Reply:
Added